infrastructure/infrastructure/src/main/java/alabno/wserver/StudentJob.java

```diff
@@ -29,4 +36,73 @@ public String readPostProcessorOutput() {
     public String getJsonLocation() {
         return jsonLocation;
     }
+
+    public void amend(String fileName, int lineno, String annType, String annotation) {
+        if ("ok".equals(annotation)) {
+            annotation = "";
+        }
+        
+        String desiredFile = toAbsolute(fileName);
+        
+        // Read the postpro file
+        String postproContent = readPostProcessorOutput();
+        
+        JsonParser parser = new JsonParser(postproContent);
+        
+        // get the annotations part
+        JsonArrayParser annotations = parser.getArrayParser("annotations");
+
+        for (JsonParser ann : annotations) {
+            
+            String aFileName = ann.getString("filename");
+            int aLineNo = ann.getInt("lineno");
+            
+            if (desiredFile.equals(aFileName) && lineno == aLineNo) {
+                JSONObject amended = ann.getObject();
+                amended.put("errortype", annType);
+                amended.put("text", annotation);
+                
+                rewriteJson(parser);
+                return;
+            }
+        }
+        
+        // If loop reaches, nothing was found
+        System.out.println("Could not find entry to be amended. Adding a new one...");
+        JSONObject newAnn = new JSONObject();
+        newAnn.put("errortype", annType);
```

Comment:
actually not, because the way 'deletion' is handled by a machine learning algorithm is actually to mark it with an 'ok'. So for the backend the deletion uses the exact same message format.